test/functional/tests/io_class/test_io_class_lvm_on_cas_dss.py

```diff
@@ -0,0 +1,151 @@
+#
+# Copyright(c) 2021 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+#
+
+import pytest
+
+from api.cas import casadm
+from api.cas.cache import Cache
+from api.cas.cache_config import CacheMode
+from api.cas.casadm_params import OutputFormat
+from api.cas.ioclass_config import IoClass
+from core.test_run import TestRun
+from storage_devices.disk import DiskType, DiskTypeSet, DiskTypeLowerThan
+from storage_devices.lvm import Lvm, LvmConfiguration
+from test_tools import fs_utils
+from test_tools.disk_utils import Filesystem
+from test_tools.fio.fio import Fio
+from test_tools.fio.fio_param import IoEngine, ReadWrite
+from test_utils.size import Size, Unit
+
+mountpoint = "/mnt/"
+io_target = "/mnt/test"
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_io_class_lvm_on_cas_dss():
+    """
+        title: IO class for CAS device behind LVM.
+        description: Validate the ability of CAS to cache IO class when CAS device is used as LVM.
+        pass_criteria:
+          - Create CAS device and LVM on top of it successfully.
+          - Loading IO class configuration successfully.
+          - Running FIO for file size from IO class from 11 to 21 successfully.
+          - Increasing proper statistics as expected.
+    """
+    with TestRun.step(f"Create CAS device."):
+
+        cache_dev, core_dev = prepare_devices()
+
+        cache = casadm.start_cache(cache_dev[0], cache_mode=CacheMode.WB, force=True)
+        core = cache.add_core(core_dev[0])
+
+    with TestRun.step("Create LVM on CAS device."):
+        lvm_filters = ["a/.*/", "r|/dev/sd*|", "r|/dev/hd*|", "r|/dev/xvd*|", "r/disk/", "r/block/",
+                       "r|/dev/nvme*|"]
+        pv_num = 1
+        vg_num = 1
+        lv_num = 1
+        cache_num = 1
+        cas_dev_num = 1
+
+        config = LvmConfiguration(lvm_filters, pv_num, vg_num, lv_num, cache_num, cas_dev_num)
+
+        lvms = Lvm.create_specific_lvm_configuration(core, config)
+        lvm = lvms[0]
+
+    with TestRun.step("Create filesystem for LVM and mount it."):
+        lvm.create_filesystem(Filesystem.ext4)
+        lvm.mount(mountpoint)
+
+    with TestRun.step("Load IO class config."):
+        cache.load_io_class("/etc/opencas/ioclass-config.csv")
+
+    with TestRun.step("Run fio for file size from IO class from 11 to 21 "
+                      "and check that correct statistics increase."):
```

Comment:
I suggest to split it to two steps to increase readability.

Reply:
That way we can have both operations in one 'for' loop and checking statistics right after each FIO operation (before it we call cache.reset_counters for the statistics check). So running the workloads first and then checking statistics would impact test scenario/flow . Or did you mean something else by the split?